apps/glusterfs/block_volume_entry.go

```diff
@@ -320,6 +320,44 @@ func (v *BlockVolumeEntry) destroyFromHost(
 	return nil
 }
 
+func (v *BlockVolumeEntry) BlockVolumeInfoFromHost(executor executors.Executor,
+	hvname string, h string) (*executors.BlockVolumeInfo, error) {
+
+	godbc.Require(hvname != "")
+	godbc.Require(h != "")
+
+	blockVolumeInfo, err := executor.BlockVolumeInfo(h, hvname, v.Info.Name)
+	if _, ok := err.(*executors.VolumeDoesNotExistErr); ok {
+		logger.Warning("Block volume %v (%v) does not exist",
+			v.Info.Id, v.Info.Name)
+		return nil, err
+	} else if err != nil {
+		logger.LogError("Unable to get block volume info: %v", err)
+		return nil, err
+	}
+	return blockVolumeInfo, nil
+}
+
+func (bv *BlockVolumeEntry) expandBlockVolumeComponents(db wdb.DB, newSize int) error {
```

Comment:
The more I look at this function the more I think we should get rid of it. It is only called in one place, the name is misleading (IMO), and it mutates the db state of the parent BHV *only* not the block volume the method is connected to.
I suggest either moving the valuable logic into a private helper function in the operation or just remove it altogether and merge the logic where needed in the operation.